### ISSUE TITLE
Accessibility roadmap

### ISSUE THREAD

**Issue by danielroe** [nuxt/nuxt#23255]
We might build these directly into nuxt or possibly via a core or adhoc module or even decoupled module (nuxt/a11y).

```[tasklist]
### Tasks
- [ ] https://github.com/nuxt/nuxt/issues/14673
- [ ] built-in [axe core](https://github.com/dequelabs/axe-core) or https://github.com/vue-a11y/vue-axe?
- [ ] devtools integration? cc: @antfu
```

**Comment by franklin-tina** [comment]
😍

**Comment by chstappert** [comment]
This is an absolutely great idea.

**Comment by mayank99** [comment]
I just wanted to chime in to say that I love this initiative. It would be great if Nuxt helped with accessible client-side routing, which is is often the biggest omission in full stack frameworks.

Route announcer is a good first step, but is generally not enough and also needs to be possible to disable because it could conflict with focus announcements. Some other things to consider adding:
- a convenient way to move focus on every navigation.
- skip links for traversing to/from the top-most layout that changed.
- documentation and examples on accessible routing patterns.

I would recommend reading Marcy Sutton's writeup on [accessible client-side routing](https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/).

Happy to discuss this further and help in any way I can :)

**Comment by danielroe** [comment]
These are great ideas to add to the list; thank you.

I think route announcer should be a built-in component like `<NuxtLoadingIndicator>` that is included in the default `app.vue` but can be removed/customised.

**Comment by Gonzo17** [comment]
Hey guys! I was thinking about writing a Nuxt module that brings axe-core to the Nuxt Devtools because I was looking for support in Vue/Nuxt and it seems that there is no actively maintained tool. Luckily I found this issue! Is someone already working on it? Can I help on this topic?

**Comment by margaretjoanmiller** [comment]
is this still being worked on? It doesn't seem like a lot of attention has been given to this issue but maybe things have been changing outside the issue?

**Comment by ultrasites** [comment]
I have written a simple a11y plugin for nuxt 3 / vue 3. Maybe it is useful for you!

[https://github.com/ultrasites/nuxt-a11y](https://github.com/ultrasites/nuxt-a11y)

**Comment by maximepvrt** [comment]
🚀 https://github.com/nuxt/a11y
